tools/devops/automation/templates/build/build.yml

```diff
@@ -49,6 +49,13 @@ steps:
   submodules: recursive
   path: s/xamarin-macios
 
+
+- ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+  - bash: |
+      git checkout $BUILD_SOURCEBRANCH
```

Comment:
What was the problem with `git reset --hard HEAD~1`?

Reply:
Chooses the wrong parent, we would need to ping to the -m 2

Reply:
For everyone else to know, since @rolfbjarne  and I talked over it on discord. Github creates two different refs when you create a PR, for example for this PR we will have:

* refs/pull/14443/merge
* refs/pull/14443/head

The first ref, is the PR merge to main, the second ref is the PR code WITHOUT merging main. VSTS by default uses */merge meaning that we always get the PR code + main. Since we moved to have different jobs that do the checkout, this means that if main gets updated, the merge ref does not contain the same code, this can mean that tests fail because for example, they got updated.

To fix this, we can do ```git reset --hard HEAD^2``` and go back to the correct commit, which is the one before the merge. The last question that we need to answers is, what happens when we push to a PR while there is a build running? This is a no issues because we cancel PR builds one a push is done as per https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#multiple-pr-updates